qcoro fails to build with gcc 11.3.0 and 12.0.1_pre20220424

```
...
[ 20%] [32m[1mLinking CXX executable await-sync-string[0m
cd /var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build/examples/basics && /usr/x86_64-pc-linux-gnu/bin/cmake -E cmake_link_script CMakeFiles/await-sync-string.dir/link.txt --verbose=1
/usr/bin/x86_64-pc-linux-gnu-c++ -march=native -O2 -pipe -Wall -Wextra -Werror -pedantic -fcoroutines -Wl,-O1 -Wl,--as-needed "CMakeFiles/await-sync-string.dir/await-sync-string_autogen/mocs_compilation.cpp.o" "CMakeFiles/await-sync-string.dir/await-sync-string.cpp.o" -o await-sync-string 
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
[ 20%] Built target await-sync-string
In file included from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.cpp:6[m[K:
/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h: In instantiation of '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> qCoro(T*, FuncPtr&&, std::chrono::milliseconds) [with T = const QProcess; FuncPtr = void (QProcess::*)(QProcess::QPrivateSignal); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<QProcess::QPrivateSignal>; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1, 1000> >][m[K':
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.cpp:23:44:[m[K   required from here
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:131:19:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[KQCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(QProcess::QPrivateSignal)>::QCoroSignal(const QCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(QProcess::QPrivateSignal)>&)[m[K'
  131 |     auto result = [01;31m[Kco_await[m[K coroSignal;
      |                   [01;31m[K^~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;36m[Knote: [m[K'[01m[KQCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(QProcess::QPrivateSignal)>::QCoroSignal(const QCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(QProcess::QPrivateSignal)>&)[m[K' is implicitly deleted because the default definition would be ill-formed:
   56 | class [01;36m[KQCoroSignal[m[K {
      |       [01;36m[K^~~~~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[Kstd::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = QTimer; _Dp = std::default_delete<QTimer>][m[K'
In file included from [01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/memory:76[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/waitoperationbase_p.h:11[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.h:7[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.cpp:5[m[K:
[01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/bits/unique_ptr.h:468:7:[m[K [01;36m[Knote: [m[Kdeclared here
  468 |       [01;36m[Kunique_ptr[m[K(const unique_ptr&) = delete;
      |       [01;36m[K^~~~~~~~~~[m[K
In file included from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.cpp:6[m[K:
/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h: In instantiation of '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> qCoro(T*, FuncPtr&&, std::chrono::milliseconds) [with T = const QProcess; FuncPtr = void (QProcess::*)(int, QProcess::ExitStatus); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<std::tuple<int, QProcess::ExitStatus> >; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1, 1000> >][m[K':
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.cpp:40:41:[m[K   required from here
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:131:19:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[KQCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(int, QProcess::ExitStatus)>::QCoroSignal(const QCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(int, QProcess::ExitStatus)>&)[m[K'
  131 |     auto result = [01;31m[Kco_await[m[K coroSignal;
      |                   [01;31m[K^~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;36m[Knote: [m[K'[01m[KQCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(int, QProcess::ExitStatus)>::QCoroSignal(const QCoro::detail::QCoroSignal<const QProcess, void (QProcess::*)(int, QProcess::ExitStatus)>&)[m[K' is implicitly deleted because the default definition would be ill-formed:
   56 | class [01;36m[KQCoroSignal[m[K {
      |       [01;36m[K^~~~~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[Kstd::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = QTimer; _Dp = std::default_delete<QTimer>][m[K'
In file included from [01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/memory:76[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/waitoperationbase_p.h:11[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.h:7[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroprocess.cpp:5[m[K:
[01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/bits/unique_ptr.h:468:7:[m[K [01;36m[Knote: [m[Kdeclared here
  468 |       [01;36m[Kunique_ptr[m[K(const unique_ptr&) = delete;
      |       [01;36m[K^~~~~~~~~~[m[K
In file included from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.cpp:7[m[K:
/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h: In instantiation of '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> qCoro(T*, FuncPtr&&, std::chrono::milliseconds) [with T = QCoro::detail::WaitSignalHelper; FuncPtr = void (QCoro::detail::WaitSignalHelper::*)(bool); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<bool>; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1, 1000> >][m[K':
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.cpp:128:29:[m[K   required from here
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:131:19:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[KQCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(bool)>::QCoroSignal(const QCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(bool)>&)[m[K'
  131 |     auto result = [01;31m[Kco_await[m[K coroSignal;
      |                   [01;31m[K^~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;36m[Knote: [m[K'[01m[KQCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(bool)>::QCoroSignal(const QCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(bool)>&)[m[K' is implicitly deleted because the default definition would be ill-formed:
   56 | class [01;36m[KQCoroSignal[m[K {
      |       [01;36m[K^~~~~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[Kstd::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = QTimer; _Dp = std::default_delete<QTimer>][m[K'
In file included from [01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/memory:76[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/task.h:12[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.h:7[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.cpp:5[m[K:
[01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/bits/unique_ptr.h:468:7:[m[K [01;36m[Knote: [m[Kdeclared here
  468 |       [01;36m[Kunique_ptr[m[K(const unique_ptr&) = delete;
      |       [01;36m[K^~~~~~~~~~[m[K
In file included from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.cpp:7[m[K:
/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h: In instantiation of '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> qCoro(T*, FuncPtr&&, std::chrono::milliseconds) [with T = QCoro::detail::WaitSignalHelper; FuncPtr = void (QCoro::detail::WaitSignalHelper::*)(long long int); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<long long int>; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1, 1000> >][m[K':
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.cpp:133:29:[m[K   required from here
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:131:19:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[KQCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(long long int)>::QCoroSignal(const QCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(long long int)>&)[m[K'
  131 |     auto result = [01;31m[Kco_await[m[K coroSignal;
      |                   [01;31m[K^~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;36m[Knote: [m[K'[01m[KQCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(long long int)>::QCoroSignal(const QCoro::detail::QCoroSignal<QCoro::detail::WaitSignalHelper, void (QCoro::detail::WaitSignalHelper::*)(long long int)>&)[m[K' is implicitly deleted because the default definition would be ill-formed:
   56 | class [01;36m[KQCoroSignal[m[K {
      |       [01;36m[K^~~~~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[Kstd::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = QTimer; _Dp = std::default_delete<QTimer>][m[K'
In file included from [01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/memory:76[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/task.h:12[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.h:7[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcoroiodevice.cpp:5[m[K:
[01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/bits/unique_ptr.h:468:7:[m[K [01;36m[Knote: [m[Kdeclared here
  468 |       [01;36m[Kunique_ptr[m[K(const unique_ptr&) = delete;
      |       [01;36m[K^~~~~~~~~~[m[K
make[2]: *** [qcoro/core/CMakeFiles/QCoro5Core.dir/build.make:121: qcoro/core/CMakeFiles/QCoro5Core.dir/qcoroprocess.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [qcoro/core/CMakeFiles/QCoro5Core.dir/build.make:93: qcoro/core/CMakeFiles/QCoro5Core.dir/qcoroiodevice.cpp.o] Error 1
In file included from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorothread.cpp:6[m[K:
/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h: In instantiation of '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> qCoro(T*, FuncPtr&&, std::chrono::milliseconds) [with T = QThread; FuncPtr = void (QThread::*)(QThread::QPrivateSignal); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<QThread::QPrivateSignal>; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1, 1000> >][m[K':
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorothread.cpp:24:39:[m[K   required from here
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:131:19:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[KQCoro::detail::QCoroSignal<QThread, void (QThread::*)(QThread::QPrivateSignal)>::QCoroSignal(const QCoro::detail::QCoroSignal<QThread, void (QThread::*)(QThread::QPrivateSignal)>&)[m[K'
  131 |     auto result = [01;31m[Kco_await[m[K coroSignal;
      |                   [01;31m[K^~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;36m[Knote: [m[K'[01m[KQCoro::detail::QCoroSignal<QThread, void (QThread::*)(QThread::QPrivateSignal)>::QCoroSignal(const QCoro::detail::QCoroSignal<QThread, void (QThread::*)(QThread::QPrivateSignal)>&)[m[K' is implicitly deleted because the default definition would be ill-formed:
   56 | class [01;36m[KQCoroSignal[m[K {
      |       [01;36m[K^~~~~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[Kstd::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = QTimer; _Dp = std::default_delete<QTimer>][m[K'
In file included from [01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/memory:76[m[K,
                 from [01m[K/usr/x86_64-pc-linux-gnu/include/qt5/QtCore/qsharedpointer_impl.h:71[m[K,
                 from [01m[K/usr/x86_64-pc-linux-gnu/include/qt5/QtCore/qsharedpointer.h:48[m[K,
                 from [01m[K/usr/x86_64-pc-linux-gnu/include/qt5/QtCore/qpointer.h:43[m[K,
                 from [01m[K/usr/x86_64-pc-linux-gnu/include/qt5/QtCore/QPointer:1[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorothread.h:7[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorothread.cpp:5[m[K:
[01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/bits/unique_ptr.h:468:7:[m[K [01;36m[Knote: [m[Kdeclared here
  468 |       [01;36m[Kunique_ptr[m[K(const unique_ptr&) = delete;
      |       [01;36m[K^~~~~~~~~~[m[K
In file included from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorotimer.cpp:6[m[K:
/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h: In instantiation of '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> qCoro(T*, FuncPtr&&, std::chrono::milliseconds) [with T = QTimer; FuncPtr = void (QTimer::*)(QTimer::QPrivateSignal); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<QTimer::QPrivateSignal>; std::chrono::milliseconds = std::chrono::duration<long int, std::ratio<1, 1000> >][m[K':
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:149:45:[m[K   required from '[01m[KQCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type::value_type> qCoro(T*, FuncPtr&&) [with T = QTimer; FuncPtr = void (QTimer::*)(QTimer::QPrivateSignal); typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type::value_type = QTimer::QPrivateSignal; typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type = std::optional<QTimer::QPrivateSignal>][m[K'
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorotimer.cpp:41:23:[m[K   required from here
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:131:19:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[KQCoro::detail::QCoroSignal<QTimer, void (QTimer::*)(QTimer::QPrivateSignal)>::QCoroSignal(const QCoro::detail::QCoroSignal<QTimer, void (QTimer::*)(QTimer::QPrivateSignal)>&)[m[K'
  131 |     auto result = [01;31m[Kco_await[m[K coroSignal;
      |                   [01;31m[K^~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;36m[Knote: [m[K'[01m[KQCoro::detail::QCoroSignal<QTimer, void (QTimer::*)(QTimer::QPrivateSignal)>::QCoroSignal(const QCoro::detail::QCoroSignal<QTimer, void (QTimer::*)(QTimer::QPrivateSignal)>&)[m[K' is implicitly deleted because the default definition would be ill-formed:
   56 | class [01;36m[KQCoroSignal[m[K {
      |       [01;36m[K^~~~~~~~~~~[m[K
[01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorosignal.h:56:7:[m[K [01;31m[Kerror: [m[Kuse of deleted function '[01m[Kstd::unique_ptr<_Tp, _Dp>::unique_ptr(const std::unique_ptr<_Tp, _Dp>&) [with _Tp = QTimer; _Dp = std::default_delete<QTimer>][m[K'
In file included from [01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/memory:76[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/task.h:12[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorotimer.h:7[m[K,
                 from [01m[K/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qcoro-0.5.0/qcoro/core/qcorotimer.cpp:5[m[K:
[01m[K/usr/x86_64-pc-linux-gnu/include/c++/11.3.0/bits/unique_ptr.h:468:7:[m[K [01;36m[Knote: [m[Kdeclared here
  468 |       [01;36m[Kunique_ptr[m[K(const unique_ptr&) = delete;
      |       [01;36m[K^~~~~~~~~~[m[K
[ 21%] [32m[1mLinking CXX executable testdbusserver[0m
cd /var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build/tests && /usr/x86_64-pc-linux-gnu/bin/cmake -E cmake_link_script CMakeFiles/testdbusserver.dir/link.txt --verbose=1
/usr/bin/x86_64-pc-linux-gnu-c++ -march=native -O2 -pipe -Wall -Wextra -Werror -pedantic -fcoroutines -Wl,-O1 -Wl,--as-needed CMakeFiles/testdbusserver.dir/testdbusserver_autogen/mocs_compilation.cpp.o CMakeFiles/testdbusserver.dir/testdbusserver.cpp.o -o testdbusserver  /usr/x86_64-pc-linux-gnu/lib/libQt5DBus.so.5.15.3 /usr/x86_64-pc-linux-gnu/lib/libQt5Core.so.5.15.3 
[ 22%] [32m[1mLinking CXX executable await-async-string[0m
cd /var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build/examples/basics && /usr/x86_64-pc-linux-gnu/bin/cmake -E cmake_link_script CMakeFiles/await-async-string.dir/link.txt --verbose=1
/usr/bin/x86_64-pc-linux-gnu-c++ -march=native -O2 -pipe -Wall -Wextra -Werror -pedantic -fcoroutines -Wl,-O1 -Wl,--as-needed "CMakeFiles/await-async-string.dir/await-async-string_autogen/mocs_compilation.cpp.o" "CMakeFiles/await-async-string.dir/await-async-string.cpp.o" -o await-async-string  /usr/x86_64-pc-linux-gnu/lib/libQt5Core.so.5.15.3 
make[2]: *** [qcoro/core/CMakeFiles/QCoro5Core.dir/build.make:149: qcoro/core/CMakeFiles/QCoro5Core.dir/qcorotimer.cpp.o] Error 1
[ 23%] [32m[1mLinking CXX executable dbusserver[0m
cd /var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build/examples/dbus/common && /usr/x86_64-pc-linux-gnu/bin/cmake -E cmake_link_script CMakeFiles/dbusserver.dir/link.txt --verbose=1
/usr/bin/x86_64-pc-linux-gnu-c++ -march=native -O2 -pipe -Wall -Wextra -Werror -pedantic -fcoroutines -Wl,-O1 -Wl,--as-needed CMakeFiles/dbusserver.dir/dbusserver_autogen/mocs_compilation.cpp.o CMakeFiles/dbusserver.dir/dbusserver.cpp.o -o dbusserver  /usr/x86_64-pc-linux-gnu/lib/libQt5DBus.so.5.15.3 /usr/x86_64-pc-linux-gnu/lib/libQt5Core.so.5.15.3 
make[2]: *** [qcoro/core/CMakeFiles/QCoro5Core.dir/build.make:135: qcoro/core/CMakeFiles/QCoro5Core.dir/qcorothread.cpp.o] Error 1
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
make[1]: *** [CMakeFiles/Makefile2:1165: qcoro/core/CMakeFiles/QCoro5Core.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 24%] [32m[1mLinking CXX static library libexamples-dbus-common.a[0m
cd /var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build/examples/dbus/common && /usr/x86_64-pc-linux-gnu/bin/cmake -P CMakeFiles/examples-dbus-common.dir/cmake_clean_target.cmake
cd /var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build/examples/dbus/common && /usr/x86_64-pc-linux-gnu/bin/cmake -E cmake_link_script CMakeFiles/examples-dbus-common.dir/link.txt --verbose=1
x86_64-pc-linux-gnu-ar qc libexamples-dbus-common.a "CMakeFiles/examples-dbus-common.dir/examples-dbus-common_autogen/mocs_compilation.cpp.o" "CMakeFiles/examples-dbus-common.dir/dbusserver.cpp.o"
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
x86_64-pc-linux-gnu-ranlib libexamples-dbus-common.a
[ 24%] Built target testdbusserver
[ 24%] Built target await-async-string
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
[ 24%] Built target examples-dbus-common
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
[ 24%] Built target dbusserver
make[2]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
[ 24%] Built target qcoro_test_dbus
make[1]: Leaving directory '/var/tmp/paludis/build/dev-libs-qcoro-0.5.0/work/qt5-build'
make: *** [Makefile:149: all] Error 2
```

Not sure if that's a problem of gcc or qcoro, but it works with gcc 10.3.0 and clang 13.0.1.

--- a/qcoro/core/qcorosignal.h
+++ b/qcoro/core/qcorosignal.h
@@ -127,8 +127,7 @@ QCoroSignal(T *, FuncPtr &&, std::chrono::milliseconds) -> QCoroSignal<T, FuncPt
 template<QCoro::detail::concepts::QObject T, typename FuncPtr>
 inline auto qCoro(T *obj, FuncPtr &&ptr, std::chrono::milliseconds timeout)
     -> QCoro::Task<typename QCoro::detail::QCoroSignal<T, FuncPtr>::result_type> {
-    QCoro::detail::QCoroSignal coroSignal(obj, std::forward<FuncPtr>(ptr), timeout);
-    auto result = co_await coroSignal;
+    auto result = co_await QCoro::detail::QCoroSignal(obj, std::forward<FuncPtr>(ptr), timeout);
     co_return std::move(result);
 }
 

